app-frontend/src/app/components/map/nodeSelector/nodeSelector.html

```diff
@@ -0,0 +1,36 @@
+<div class="node-header-colorblock"
+     ng-class="{'node-type-input': $ctrl.selectedType === 'src',
+                'node-type-function': $ctrl.selectedType === 'function'}"
+></div>
+<div class="node-selector-dropdown">
+  <button
+    class="btn btn-transparent btn-text"
+    ng-if="!$ctrl.showSearch"
+    ng-click="$ctrl.startSelecting()"
```

Comment:
This works pretty well, but there was one thing that struck me as odd: when I click in the text field itself, it disappears and just switches back to whatever item I had selected before clicking on it. This makes it difficult to interact with the text field in the ways I expect to be available, like selecting text or clicking to move the cursor. I think that clicking in the text field when the dropdown is open shouldn't close it.